charts/consul/templates/server-statefulset.yaml

```diff
@@ -238,6 +238,7 @@ spec:
         volumeMounts:
           - name: extra-config
             mountPath: /consul/extra-config
+        {{- include "consul.restrictedSecurityContext" . | nindent 8 }}
```

Comment:
Please add a bats test to test consul.restrictedSecurityContext. Even though the definition is in _helpers we still want to test it at least once and server-statefulset is one of the more common resources to test helpers in. No need to add bats tests for all the other files that use consul.restrictedSecurityContext.

Reply:
I updated the bats tests for server-statefulset